pa4/checker-pa4.ml

```diff
@@ -258,6 +264,22 @@ let add_attribute_types o input_class = begin
 end;;
 
 
+
+let add_method_types meth_env input_class = begin
```

Comment:
I'm not a fan of `m_e` as a variable name, but if we're going to use it below to describe the method environment, we should use it here as well, I believe. I do feel saddened that we can have a `meth_env` here, but I think it's a sacrifice we will have to make.